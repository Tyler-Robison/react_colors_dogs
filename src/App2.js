import React, { useState } from 'react'
import './App.css';
import RouteList from './RouteList';
import Nav from './Nav';
import colorsArr from './colors';
import useLocalStorage from './hooks/useLocalStorage';

function App2() {

  const [colors, setColors, resetColors] = useLocalStorage('colors', colorsArr)

  return (
    <div className="App">
      <h1>app 2</h1>
      <Nav colors={colors} />
      <RouteList setColors={setColors} colors={colors} />
      <button onClick={resetColors}>Reset Colors</button>
    </div>
  );
}

export default App2;