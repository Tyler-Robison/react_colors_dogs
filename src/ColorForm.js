import React, { useState } from 'react'
import { useNavigate } from "react-router-dom";

const ColorForm = ({ setColors, colors }) => {

    const navigate = useNavigate();

    const INITIAL_STATE = {
        color: ''
    }

    const [formData, setFormData] = useState(INITIAL_STATE)

    const handleChange = (e) => {
        const { name, value } = e.target
        setFormData(formData => ({
            ...formData, [name]: value
        }))
    }

    const handleSubmit = (e) => {
        e.preventDefault()
        const color = formData.color

        // prevent changing to invalid color
        if (!CSS.supports('color', color)) {
            alert(`${color} is invalid`);
            return
        }

        setColors([...colors, color])
        setFormData(INITIAL_STATE)
        navigate('/')
    }

    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="color-input">New Color</label>
            <input
                id='color-input'
                type='text'
                name='color'
                placeholder="Enter Color Here"
                onChange={handleChange}
            />
            <button>Add color</button>

        </form>
    )
}

export default ColorForm;