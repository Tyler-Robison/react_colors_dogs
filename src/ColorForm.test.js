import React from "react";
import { render, fireEvent } from "@testing-library/react";
import ColorForm from "./ColorForm";
import { MemoryRouter } from "react-router-dom";
import App2 from "./App2";

it("renders without crashing", function () {
    <MemoryRouter initialEntries={['/']}>
        <ColorForm />
    </MemoryRouter>
});

it("matches snapshot", function () {
    const container = render(
        <MemoryRouter initialEntries={['/colors/new']}>
            <ColorForm />
        </MemoryRouter>
    );
    expect(container.asFragment()).toMatchSnapshot();
});