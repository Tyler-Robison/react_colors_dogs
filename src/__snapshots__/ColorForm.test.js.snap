// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="color-input"
    >
      New Color
    </label>
    <input
      id="color-input"
      name="color"
      placeholder="Enter Color Here"
      type="text"
    />
    <button>
      Add color
    </button>
  </form>
</DocumentFragment>
`;
